lib/global-admin/addon/components/form-global-roles/component.js

```diff
@@ -1,83 +1,76 @@
 import Component from '@ember/component'
 import { all as PromiseAll } from 'rsvp';
 import { inject as service } from '@ember/service';
-import { get, set, observer } from '@ember/object';
+import { get, set, setProperties } from '@ember/object';
 import layout from './template';
-import { on } from '@ember/object/evented';
-
-const USER = 'user';
-const ADMIN = 'admin';
-const CUSTOM = 'custom';
+import $ from 'jquery';
+
+const USER               = 'user';
+const ADMIN              = 'admin';
+const BASE               = 'user-base';
+const LOGIN_ACCESS_ROLES = [
+  {
+    name:                      USER,
+    translationKey:            'formGlobalRoles.mode.user.label',
+    descriptionTranslationKey: 'formGlobalRoles.mode.user.detail',
+  },
+  {
+    name:                      ADMIN,
+    translationKey:            'formGlobalRoles.mode.admin.label',
+    descriptionTranslationKey: 'formGlobalRoles.mode.admin.detail',
+  },
+  {
+    name:                      BASE,
+    translationKey:            'formGlobalRoles.mode.userBase.label',
+    descriptionTranslationKey: 'formGlobalRoles.mode.userBase.detail',
+  },
+];
 
 export default Component.extend({
   settings:    service(),
   globalStore: service(),
 
   layout,
-  user:        null,
-
-  mode:        null,
-  allRoles:    null,
-  custom:      null,
-
-  _boundSave:  null,
+  user:            null,
+  mode:            null,
+  allRoles:        null,
+  _boundSave:      null,
+  _boundValidator: null,
 
   init() {
     this._super(...arguments);
 
-    const all       = set(this, 'allRoles', get(this, 'globalStore').all('globalRole'));
-    const user      = get(this, 'user');
-    const hasCustom = !!get(user, 'globalRoleBindings').map((x) => get(x, 'globalRole')).findBy('isCustom');
-    const current   = get(user, 'globalRoleBindings');
+    set(this, 'allRoles', this.globalStore.all('globalRole'));
 
-    let mode        = CUSTOM;
+    this.initRoles();
 
-    if ( get(user, 'hasAdmin') ) {
-      mode = ADMIN;
-    } else if ( (!get(user, 'id') || get(user, 'hasUser')) && !hasCustom ) {
-      mode = USER;
+    if (this.mode === 'new') {
+      this.populateDefaultRoles();
```

Comment:
Based on the naming I suspect this makes more sense to go into initRoles().

Reply:
Moved. I had it there originally and waffled back and forth its placement. Your comment validates my original thought that it belongs in initRoes.